dbos/_outcome.py

```diff
@@ -28,7 +30,7 @@ def retry(
         exceeded_retries: Callable[[int], BaseException],
     ) -> "Outcome[T]": ...
 
-    def intercept(self, interceptor: Callable[[], Optional[T]]) -> "Outcome[T]": ...
+    def intercept(self, interceptor: Callable[[], Optional[str]]) -> "Outcome[T]": ...
```

Comment:
I'm not sure if this is the best solution. Basically I moved deserialization out of the interceptor. However, it means the interceptor will always return a string.

Reply:
Another option is to change the return type to some other protocol, like `{ "output": ...}` Then if you get `None` that means nothing.